Complete feature

jdavis218 opened the pull request:
Hey Scott and Alex, 

I had a great time working on this! I implemented the features as detailed in Scott's email. There were some slight discrepancies between what was already provided in the code and what was represented in the screenshot in the design folder (i.e. adding a description for each task). So far, I've just added the features specified in the email, but please let me know if you'd like me to add any additional features or make any changes at all. Please also let me know if you have any questions.

Looking forward to hearing from you!

alexpeachey commented:
Before diving into the code, I ran the application and it looks pretty nice. I liked the responsiveness when I changed shape/size of the browser and the visual were nicely done.
I did notice a couple things:
1. The graphs do not represent the state of the application when first loaded. I needed to interact for them to render appropriately (create a new task or change state on a task by clicking a button)
2. When adding a new task I had to explicitly click the button, enter wasn't good enough. That wasn't a requirement and definitely not a "problem" for not having that work, just something a little clunky usability-wise.

alexpeachey commented on src/components/Graph.js:
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { Circle } from "rc-progress";
+
+const Graph = ({ count, progress, color, label }) => {
```
I like the use of a simple stateless component for `Graph`

alexpeachey commented on src/components/Graphs.js:
```diff
@@ -0,0 +1,24 @@
+  if (totalTasks > 0) {
+    toDoCount = Math.round((tasks.todo / totalTasks) * 100);
+    inProgressCount = Math.round((tasks.inProgress / totalTasks) * 100);
+    completeCount = Math.round((tasks.complete / totalTasks) * 100);
```
These might have better names like `toDoPercentage` or something. Count implies they are something different than what they are.

jdavis218 commented:
Hey Alex,

Thank you for the feedback! And thank you for catching that initial rendering of the graphs issue - when I altered the initial state object, I failed to update the total, todo, and in progress counts. This is now fixed, and I've also updated the variable names for the percentage calculations, as well as changed the submit button from an onClick event to an onSubmit event for more flexibility when entering data. Let me know if you have any additional questions or would like me to change or add any other features.

Thanks!
